### ISSUE TITLE
Dynamically turn on and off displays

### ISSUE THREAD

**Issue by Sefff-120** [swaywm/sway#7809]
Add an option to automatically enable displays when they are connected/lid is opened, and disable when disconnected/lid is closed. I have a laptop connected to a dock, and usually disable the laptop screen. But when I disconnect it from the dock, the screen stays off, and I cannot do anything (often I have to reboot the laptop). It would be nice if it would automatically enable the screen like other wayland compositors (gnome).

**Comment by sigboe** [comment]
Yes it would be nice if sway could do this, in the meantime you can use kanshi to automate this
https://sr.ht/~emersion/kanshi/

I have this in my sway config ( I don't know if the ampersand is needed
```
    exec kanshi &
    exec_always kanshictl reload 
```
I have kanshi installed via `sudo pacman -S kanshi` package name on other distros may be different 

In my `~/.config/kanshi/config` I have this in the beginning, keep in mind that the name like eDP-1 needs to be what `swaymsg -t get_outputs` says. Keep in mind I have enable there, I found sometimes when I disconnect maybe it was during my laptop was closed, the screen didn't enable when I opened the lid, and this fixed that for me. Maybe for you too since you disable the screen

```
profile {
	output eDP-1 enable position 0,0 scale 2
}
```

Another profile object for example
```
profile {
	output "GIGA-BYTE TECHNOLOGY CO., LTD. M34WQ 0x000006E2" mode 3440x1440 position 0,0
	output eDP-1 position 3440,820 scale 2.5
}
```
Kanshi enables the profile, if you have a profile that fits the exact names for descriptions for the screens, example in this situation, if I only have the internal screen active, it enables the first profile because that fits having only the screens in the profile. When I connect my gigabyte monitor, the next profile is enabled because it recognizes that I have those two monitors and I have a profile with those two monitors.

Whenever I connect a new screen, Idk but it automatically connects in sway, and Idk but I always get the 2.5 scale on my laptop screen which is fine for me. but I use an app like wdisplays to configure the screen how I like, sadly it doesn't have a save to kanshi button yet, but last I checked there was an open pull request with this very feature. But I just for now have copied a profile, changed the name, and copied out values from the number boxes in wdisplays. If you want to just compile wdisplays with that feature, here is the pull request https://github.com/artizirk/wdisplays/pull/11
